### ISSUE TITLE
Creating a Mixout module for XLNet

### ISSUE THREAD

**Issue by fteufel** [bloodwass/mixout#3]
Hi,

I wanted to apply mixout to XLNet (HuggingFace implementation, https://huggingface.co/transformers/_modules/transformers/modeling_xlnet.html#XLNetModel). However, unlike Bert, with MixLinear I can only apply it to half the dropouts - it only works for `XLNetFeedForward`, not for `XLNetRelativeAttention`.

```
XLNetLayer(
        (rel_attn): XLNetRelativeAttention(
          (layer_norm): LayerNorm((1024,), eps=1e-12, elementwise_affine=True)
          (dropout): Dropout(p=0.1, inplace=False)
        )
        (ff): XLNetFeedForward(
          (layer_norm): LayerNorm((1024,), eps=1e-12, elementwise_affine=True)
          (layer_1): Linear(in_features=1024, out_features=4096, bias=True)
          (layer_2): Linear(in_features=4096, out_features=1024, bias=True)
          (dropout): Dropout(p=0.1, inplace=False)
        )
        (dropout): Dropout(p=0.1, inplace=False)
)
```

Dropout is applied after the computation of attention scores. Therefore I would have to implement a `MixXLNetRelativeAttention` module, where, in the forward call, i wrap all references to weights of the layer (`self.q, self.k, self.v, self.o, self.r self.r_r_bias, self.r_s_bias, self.r_w_bias, self.seg_embed`) with `mixout`?

I'm just asking because it appears much more complicated than for Bert - but I get it, Bert seems to be using `nn.Linear` everywhere rather than raw `nn.Parameter`.

Also, in `MixLinear` you do not wrap the bias, only the weights. Would you recommend following this approach here?

**Comment by bloodwass** [comment]
The first priority of our paper was comparing mixout with dropout in BERT, so we only applied mixout to parameters affected by dropout (and the bias of nn.Linear is not dropped). In your case, I think you'd like to avoid too much deviation from pre-trained parameters including bias, and you believe that it would be helpful. Then, go ahead! We don't need to apply mixout only for weights.
